extensions/amp-apester-media/0.1/monetization/consent-util.js

```diff
@@ -0,0 +1,78 @@
+/* eslint-disable google-camelcase/google-camelcase */
+
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {dict} from '../../../../src/utils/object';
+import {
+  getConsentPolicyInfo,
+  getConsentPolicyState,
+} from '../../../../src/consent';
+
+import {dev} from '../../../../src/log';
+
+import {CONSENT_POLICY_STATE} from '../../../../src/consent-state';
+
+const TAG = 'amp-apester-media';
+
+const AWAIT_TIME_OUT_FOR_RESPONSE = 3000;
+
+const awaitPromiseTimeout = () => {
+  return new Promise(resolve => {
+    setTimeout(() => {
+      resolve([CONSENT_POLICY_STATE.UNKNOWN, undefined]);
+    }, AWAIT_TIME_OUT_FOR_RESPONSE);
+  });
+};
+/**
+ * @param {AmpElement} apesterElement
+ * @return {Promise<!JsonObject>}
+ * */
+export function getConsentData(apesterElement) {
+  const consentStatePromise = getConsentPolicyState(apesterElement).catch(
+    err => {
+      dev().error(TAG, 'Error determining consent state', err);
+      return CONSENT_POLICY_STATE.UNKNOWN;
+    }
+  );
+  const consentStringPromise = getConsentPolicyInfo(
+    apesterElement,
+    'default'
+  ).catch(err => {
+    dev().error(TAG, 'Error determining consent string', err);
+    return undefined;
+  });
+  const consentDataPromise = Promise.all([
+    consentStatePromise,
+    consentStringPromise,
+  ]);
+  return Promise.race([consentDataPromise, awaitPromiseTimeout()]).then(
```

Comment:
AMP has a helper for this: `timeoutPromise`: https://github.com/ampproject/amphtml/blob/12e4ada55c87979b1486e5f5d734adb193f6b4cd/src/service/timer-impl.js#L127-L138

It allows you to specify a Promise to race against, but that makes handling the timeout and recovering a bit messy, but you can still use it for the timeout part.

```javascript
const timeout = timeoutPromise(delay).catch(e => {
  // Fallback data for timeout.
  return ...;
});
return Promise.race([consentDataPromise, timeout]).then(...);
```

Reply:
@sparhami Thanks for that, If it's ok with you I'm strongly prefer not to change it, I feel like now it will be more readable for us at the future.